sdk/identity/azure-identity-v2/src/main/java/com/azure/v2/identity/implementation/client/BrokeredAuthCache.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.v2.identity.implementation.client;
+
+import com.azure.v2.core.credentials.TokenRequestContext;
+import com.azure.v2.identity.implementation.models.MsalAuthenticationAccount;
+import com.azure.v2.identity.implementation.models.MsalToken;
+import com.azure.v2.identity.implementation.models.PublicClientOptions;
+import com.azure.v2.identity.models.AuthenticationRecord;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import io.clientcore.core.credentials.oauth.AccessToken;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Brokered Auth Cache offers APIs to manage and read cache for Public client brokered auth flows.
+ */
+public class BrokeredAuthCache {
```

Comment:
I worry we're overloading "brokered" here. `MsalAccountCache` or `MsalAuthenticationAccountCache`?